src/libraries/System.Private.CoreLib/src/System/TimeSpan.cs

```diff
@@ -271,6 +271,22 @@ public static TimeSpan FromDays(double value)
             return Interval(value, TicksPerDay);
         }
 
+        public static TimeSpan FromDays(int days)
+        {
+            return new TimeSpan((long)days * TicksPerDay);
+        }
+
+        public static TimeSpan FromDays(int days, int hours = 0, long minutes = 0, long seconds = 0, long milliseconds = 0, long microseconds = 0)
```

Comment:
> public static TimeSpan FromDays(int days, int hours = 0, long minutes = 0, long seconds = 0, long milliseconds = 0, long microseconds = 0) [](http://example.com/codeflow?start=7&length=139)

Please look at the code of the constructor `public TimeSpan(int days, int hours, int minutes, int seconds, int milliseconds, int microseconds)`. It has some logic to ensure no overflow by checking the total max microseconds. I suggest you refactor this code into its own method and then call it from here and from the constructor.

Reply:
wait, we are using long and not int for the parameters. I think we need to implement this to ensure there is no overflow will occur.

Reply:
@tarekgh I'm struggling to see the benefit of manually checking for overflows given that `long.MaxValue` represents the upper limit for TimeSpan's length. Any computation exceeding this limit inherently triggers an `OverflowException`. Implementing explicit checks seems to introduce additional CPU overhead, thus potentially reducing performance.

Here are some benchmarks:
```
| Method                            | Input               | Mean          | Error      | StdDev     | Gen0   | Allocated |
|---------------------------------- |-------------------- |--------------:|-----------:|-----------:|-------:|----------:|
| DefaultOverflowException          | 9223372036854775807 |     0.0192 ns |  0.0177 ns |  0.0165 ns |      - |         - |
| ManualOverflowChecksThenException | 9223372036854775807 | 3,188.9095 ns | 42.1336 ns | 39.4118 ns | 0.0229 |     240 B |
```

Reply:
> Any computation exceeding this limit inherently triggers an OverflowException

No. this will trigger it if the operation is `checked` which we don't enable by default. https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/checked-and-unchecked

Reply:
> > Any computation exceeding this limit inherently triggers an OverflowException
> 
> No. this will trigger it if the operation is `checked` which we don't enable by default. https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/checked-and-unchecked

I see, and what are the implications of having the `checked` keyword being used explicitly in there?

Reply:
That is what I was trying to point at in my comment https://github.com/dotnet/runtime/pull/94143#discussion_r1375532302